internal/bminventory/inventory.go

```diff
@@ -2791,6 +2819,12 @@ func (b *bareMetalInventory) ResetCluster(ctx context.Context, params installer.
 		return installer.NewResetClusterInternalServerError().WithPayload(common.GenerateError(http.StatusInternalServerError, err))
 	}
 
+	if err := tx.Model(&common.Cluster{}).Where("id = ?", params.ClusterID).Update(
```

Comment:
I think this should be on the state-machine like we reset the hosts' data on `PostRegisterHost`